Made changes using test_board

glennqhurd opened the pull request:
Used the file test_board from Lyman Hurd to mess with the layout using CSS.  lyngk_board.js is the slightly modified version of previous Javascript code.

glennqhurd committed:
Made changes using test_board

Used the file test_board from Lyman Hurd to mess with the layout using CSS.  lyngk_board.js is the slightly modified version of previous Javascript code.

glennqhurd committed:
Added new canvases to page

 Added three new canvases to the page: Unclaimed, player 1 claimed, and player 2 claimed.  It will delineate what colors are claimed as part of the game which should allow the user to more easily see the differences since it's colored images instead of text.

glennqhurd committed:
Made further changes to canvases

Added hidden HTML code to determine what pieces were shown as claimed by players using a Javascript function to read those HTML code elements and draw the pieces accordingly.

glennqhurd committed:
Made further layout changes

Reduced size of canvases so there is room for up to 2 since the players only claim two colors.  Also changed the spacing of the unclaimed colors.  Then rearranged the layout and added scroll attribute to the table.

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+    '-': "Empty"
+  };
+  var colorArray = ['R', 'K', 'B', 'I', 'G'];
+  var kTestString1 = 'KRI|-|G|-|GR|-|IG|KJ|-|-|-|-|RK|-|J|KB|I|-|-|-|-|K|G|-|I|GBI|-|I|GIJR|B|-|RB|RKBI|G|RK|-|GB|-|-|RB|-|BK|-';
```
Is this still necessary?

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+  var colorArray = ['R', 'K', 'B', 'I', 'G'];
+  var kTestString1 = 'KRI|-|G|-|GR|-|IG|KJ|-|-|-|-|RK|-|J|KB|I|-|-|-|-|K|G|-|I|GBI|-|I|GIJR|B|-|RB|RKBI|G|RK|-|GB|-|-|RB|-|BK|-';
+
+  var kTestString = 'G|B|K|K|R|K|I|I|K|G|J|K|K|B|I|B|G|R|G|K|R|G|K|J|B|R|R|B|G|J|G|I|G|B|R|B|I|I|R|B|I|R|I|';
```
Same comment...

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+      case 2:
+        boardLabel = "player2canvas";
+        break;
+    }
```
You can use a ternary operator (?:) just like C++ to reduce nine lines of code to 1.

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+    ctx.arc(40, piecePosition * 62, 10, 0, 2*Math.PI);
+    ctx.fillStyle = colorValues[letterToColor[color]];
+    ctx.fill();
+    ctx.stroke();
```
You have the act of drawing a disk at least three different times with different parameters.  Time to use a shared function.

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+    var player1ctx = claimedBoard1.getContext("2d");
+    player1ctx.clearRect(0, 0, claimedBoard1.width, claimedBoard1.height);
+    var claimedBoard2 = document.getElementById("player2canvas");
+    var player2ctx = claimedBoard2.getContext("2d");
```
A lot of duplicate code between claimed board 1 and 2!  Same comment as above.  As a general rule, try to avoid repeating yourself!

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+var lyngkModule = (function() {
+  my = {};
+  var verticalX = [1, 2, 3, 4, 5, 6, 7];
```
These names could use work.  Try to find a more intuitive description without getting wordy!

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+    }
+  }
+
+  function stackGamePiece(xcenter, ycenter, color, stackCount, ctx) {
```
Try to name functions what they do.  Perhaps in this case, "drawStackedGamePiece"?

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+    return inputArray;
+  }
+
+  function translateBoardString(boardString, ctx) {
```
Are you actually translating a board string or drawing it :-)?

lymanhurd commented on style.css:
```diff
@@ -1,7 +1,84 @@
+  left: 20%;
+}
+
+#unclaimedCanvas {
```
Try allowing the divs to float.  The layout you are trying to get has the "unclaimed colors" on the left of the "board" and then on the right we have a stack of "board title", "player to move", "move list", "first player's colors", "second player's colors".  I still don't see where "number of stacks claimed" went...

https://www.w3schools.com/cssref/pr_class_float.asp

lymanhurd commented on test_board.html:
```diff
@@ -0,0 +1,58 @@
+        </h3>
+      </div>
+      <div id ='summaryandmoves'>
+        <h1>Summary of Lyngk Board 191</h1>
```
For consistency with the existing PHP games, let's write: "Lyngk game 191" instead of "Summary of...".  See for example:

http://www.gamerz.net/pbmserv/Zertz/Zertz.php?2000

lymanhurd commented on test_board.html:
```diff
@@ -0,0 +1,58 @@
+        <div id='boardstring' class='hiddenHTML'>|B|K|KGR||K|I|I|K|G|J|K|K|B|I|B|G|R|G|K|R|G|K|J|B|R|R|B|G|J|G|I|G|B|R|B|I|I|R|B|I|R|I|</div>
+      </div>
+      <canvas width=400 height=400 id="lyngkboard">
+        Your browser does not support the HTML5
```
I know this sentence got copied but it does not make complete sense.  How about "Your browser does not support HTML5 canvases."

lymanhurd commented on test_board.html:
```diff
@@ -0,0 +1,58 @@
+          Your browser does not support the HTML5
+      </canvas>
+      <div id='player1div'>
+        <p>Claimed by Player 1:</p>
```
Insert player's name (e.g., p1).  For testing, I suggest we use two player names of differing lengths.

lymanhurd commented on test_board.html:
```diff
@@ -0,0 +1,58 @@
+      </canvas>
+      <script src="./lyngk_board.js"></script>
+</body>
+</html>
```
End files with a carriage return.

lymanhurd replied:
Add carriage return.

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+      for (var j = 0; j < boardArray[i].length; j++) {
+        var boardIndex = nodeList[i];
+        var color = letterToColor[boardArray[i][j]];
+        if (color != "Empty") {
```
Small point, but if the color == empty you can break out of the innermost loop...

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+  }
+
+  function shuffleArray(input) {
+    var inputArray = input;
```
Not sure why you are assigning.  This does not make a new copy.

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+  var yOffset = 25;
+  var padding = 25;
+  var boardPosition = {
+    A1: [0, 6],
```
Can you put all the A's on a line, B's etc?  The longest lines may need breaking into two; hard to tell by inspection.

lymanhurd commented on lyngk_board.js:
```diff
@@ -0,0 +1,210 @@
+      Green: 8,
+      Black: 8,
+      White: 3
+    };
```
Again a ternary statement can make this much shorter since the logic amounts to if Color == White return 3 else 8...  But, come to think of it, why is this needed?

lymanhurd commented on test_board.html:
```diff
@@ -0,0 +1,58 @@
+      </canvas>
+      <div id='player2div'>
+        <p>Claimed by Player 2:</p>
+      </div>
```
This <div> should include the title and the canvas.  Then there will be no need to add separate positioning for both.  They are a unit.

lymanhurd commented on test_board.html:
```diff
@@ -0,0 +1,58 @@
+      <canvas width=80 height=400 id="unclaimedCanvas">
+          Your browser does not support the HTML5
+      </canvas>
+      <div id='player1div'>
```
See comment below...

lymanhurd commented on test_board.html:
```diff
@@ -0,0 +1,58 @@
+	         <tr><td>11 a1-b3</td><td>12 R,b4-b3</td></tr>
+          </table>
+        </div>
+        <div id='player1claimed' class='hiddenHTML'>RK</div>
```
Not certain if there is a good argument for putting the hidden HTML fields in the body vs the header.

lymanhurd reviewed (changes requested):
Hi.  Here are some preliminary suggestions.

glennqhurd committed:
Made changes based on code review

Made changes to layout based on code review including grouping label and canvas in divs for both player 1 and player 2 claimed areas.

glennqhurd committed:
Made changes to HTML based on suggestions

Followed suggestions by Lyman Hurd for revising the HTML.

glennqhurd committed:
Added updated files to html folder

Added files to HTML folder to better match parent directory.
